jsk_2014_picking_challenge/scripts/object_verification.py

```diff
@@ -33,24 +38,31 @@ def _cb_bof(self, msg):
         objects_proba = dict(zip(msg.candidates, msg.probabilities))
         self.bof_data = (msg.header.stamp, objects_proba)
 
+    def _cb_cfeature(self, msg):
+        objects_proba = dict(zip(msg.candidates, msg.probabilities))
+        self.cfeature = (msg.header.stamp, objects_proba)
+
     def spin_once(self):
-        if self.bof_data is None:
+        if self.bof_data is None or self.cfeature is None:
             return
-        stamp, objects_proba = self.bof_data
+        stamp, bof_objects_proba = self.bof_data
+        stamp, cfeature_objects_proba = self.cfeature
+
         target_bin = rospy.get_param('/target', None)
         if target_bin is None:
             return
-        target_object = self.work_order[target_bin]
         candidates = self.bin_contents[target_bin]
-        proba = [(c, objects_proba[c]) for c in candidates]
+        proba = [(c, bof_objects_proba[c] + cfeature_objects_proba[c]) for c in candidates]
+
         matched = sorted(proba, key=lambda x: x[1])[-1][0]
         # compose msg
         msg = ObjectRecognition()
         msg.header.stamp = stamp
         msg.matched = matched
-        msg.probability = objects_proba[matched]
+        msg.probability = max([prob for c, prob in proba])
```

Comment:
I thought this can be `msg.probability = dict(proba)[matched]` .
but it's okay.